selene_sdk/samplers/multi_sampler.py

```diff
@@ -0,0 +1,324 @@
+                return data.numpy(), targets.numpy()
+            except StopIteration:
+                #If DataLoader iterator reaches its length, reinitialize
+                self._iterators[mode] = iter(self._dataloaders[mode])
+                data, targets = next(self._iterators[mode])
+                return data.numpy(), targets.numpy()
+
+    def get_data_and_targets(self, batch_size, n_samples, mode=None):
+        """
+        This method fetches a subset of the data from the sampler,
+        divided into batches. This method also allows the user to
+        specify what operating mode to run the sampler in when fetching
+        the data.
+
+        Parameters
+        ----------
+        batch_size : int
+            The size of the batches to divide the data into.
+        n_samples : int
+            The total number of samples to retrieve.
+        mode : str, optional
+            Default is None. The operating mode that the object should run in.
+            If None, will use the current mode `self.mode`.
+        """
+        if self._samplers[mode if mode else self.mode]:
+            return self._samplers[mode if mode else self.mode].get_data_and_targets(
+                batch_size, n_samples)
+        else:
+            self._set_batch_size(batch_size, mode=mode)
+            data_and_targets = []
+            targets_mat = []
+            count = batch_size
+            while count < n_samples:
+                data, tgts = self.sample(batch_size=batch_size, mode=mode)
+                data_and_targets.append((data, tgts))
+                targets_mat.append(tgts)
+                count += batch_size
+            remainder = batch_size - (count - n_samples)
+            data, tgts = self.sample(batch_size=remainder)
+            data_and_targets.append((data, tgts))
+            targets_mat.append(tgts)
+            targets_mat = np.vstack(targets_mat)
+            return data_and_targets, targets_mat
+
+    def get_validation_set(self, batch_size, n_samples=None):
+        """
+        This method returns a subset of validation data from the
+        sampler, divided into batches.
+
+        Parameters
+        ----------
+        batch_size : int
+            The size of the batches to divide the data into.
+        n_samples : int, optional
+            Default is None. The total number of validation examples to
+            retrieve. Handling for `n_samples=None` should be done by
+            all classes that subclass `selene_sdk.samplers.Sampler`.
+
+        """
+        return self.get_data_and_targets(
+            batch_size, n_samples, mode="validate")
+
+    def get_test_set(self, batch_size, n_samples=None):
+        """
+        This method returns a subset of testing data from the
+        sampler, divided into batches.
+
+        Parameters
+        ----------
+        batch_size : int
+            The size of the batches to divide the data into.
+        n_samples : int or None, optional
```

Comment:
is this the same as `get_validation_set` ('Handling for `n_samples=None` should be done by all classes that subclass...')

Reply:
yes it should ... . Also I just realized that DataLoader does not support `get_validation_set` and `get_test_set` yet because it does not have `get_data_and_targets`. Actually I think it had that implemented at some point but deleted - should I add it back (just by passing it to the sampler that SamplerDataLoader wraps over, so no multiple processing)?